source/server/server.cc

```diff
@@ -301,6 +301,10 @@ void InstanceImpl::updateServerStats() {
       enumToInt(Utility::serverState(initManager().state(), healthCheckFailed())));
   server_stats_->stats_recent_lookups_.set(
       stats_store_.symbolTable().getRecentLookups([](absl::string_view, uint64_t) {}));
+
+  // Update certificate stats for each context.
+  ssl_context_manager_->iterateContexts(static_cast<std::function<void(Envoy::Ssl::Context&)>>(
```

Comment:
ditto

Reply:
The reason I have a static_cast of a std::function is due to ambiguity of iterateContexts since I have overloaded the `iterateContexts` that also takes in a `std::function<void(Context&)>`. which has non-const Context (see context_manager.h). 

The reason I needed to overloaded it is because I use it here and `context.updateCertStats()` is not const.

Reply:
Can I suggest you instead rename the overload? Forcing this usage is a bit awkward.